src/main/scala/format/Forsyth.scala

```diff
@@ -125,6 +125,7 @@ object Forsyth {
         }
       case word => word -> None
     }
+    if (pockets.isDefined && !variant.crazyhouse) return None
```

Comment:
`return` is a naughty word, at least in my book. It skips execution and breaks the flow. I never use it in scala projects.

Instead, `if (...) None else { ... }` should work